crates/services/relayer/src/ports/tests.rs

```diff
@@ -24,49 +29,93 @@ fn test_insert_messages() {
 
     let mut m = Message::default();
     m.set_amount(10);
-    m.set_da_height(12u64.into());
+    m.set_da_height(same_height.into());
     let mut m2 = m.clone();
     m2.set_nonce(1.into());
     assert_ne!(m.id(), m2.id());
-    let messages = [m, m2];
-    db.insert_messages(&12u64.into(), &messages[..]).unwrap();
+    let messages = [m.into(), m2.into()];
+    db.insert_events(&same_height.into(), &messages[..])
+        .unwrap();
 }
 
 #[test]
 fn insert_always_raises_da_height_monotonically() {
-    let messages: Vec<_> = (0..10)
+    // Given
+    let same_height = 12u64.into();
+    let events: Vec<_> = (0..10)
         .map(|i| {
             let mut message = Message::default();
             message.set_amount(i);
-            message.set_da_height(i.into());
+            message.set_da_height(same_height);
             message
         })
         .map(Into::into)
         .collect();
 
     let mut db = MockStorage::default();
-    db.expect_insert::<Messages>().returning(|_, _| Ok(None));
+    db.expect_insert::<History>().returning(|_, _| Ok(None));
     db.expect_insert::<RelayerMetadata>()
         .once()
-        .withf(|_, v| **v == 9)
+        .withf(move |_, v| *v == same_height)
         .returning(|_, _| Ok(None));
     db.expect_commit().returning(|| Ok(()));
     db.expect_get::<RelayerMetadata>()
         .once()
         .returning(|_| Ok(None));
 
+    // When
     let mut db = db.into_transactional();
-    db.insert_messages(&9u64.into(), &messages[5..]).unwrap();
+    let result = db.insert_events(&same_height, &events);
 
-    let mut db = MockStorage::default();
-    db.expect_insert::<Messages>().returning(|_, _| Ok(None));
-    db.expect_commit().returning(|| Ok(()));
-    db.expect_get::<RelayerMetadata>()
-        .once()
-        .returning(|_| Ok(Some(std::borrow::Cow::Owned(9u64.into()))));
+    // Then
+    assert!(result.is_ok());
+}
 
+#[test]
+fn insert_fails_for_messages_with_different_height() {
+    // Given
+    let last_height = 1u64;
+    let events: Vec<_> = (0..=last_height)
+        .map(|i| {
+            let mut message = Message::default();
+            message.set_da_height(i.into());
+            message.set_amount(i);
+            message.into()
+        })
+        .collect();
+
+    let db = MockStorage::default();
+
+    // When
     let mut db = db.into_transactional();
-    db.insert_messages(&5u64.into(), &messages[..5]).unwrap();
+    let result = db.insert_events(&last_height.into(), &events);
+
+    // Then
+    assert!(result.is_err());
```

Comment:
Hmm. I would say I'd prefer this checks that it's the correct Error variant, but `StorageError` is pretty type-erased :(

Reply:
Actually. I've been thinking about this a bit. I think it kinda makes sense that `StorageError` is somewhat erased, since it's up the implementer to create their own implementation-specific errors. I think it is still possible to define the port errors with some domain specificity. That probably falls out of the scope of this PR though.